app/controllers/webapi_controller.rb

```diff
@@ -0,0 +1,29 @@
+class WebapiController < ApplicationController
+  include ActionView::Helpers::AssetUrlHelper
+
+  #history of all games
+  def history
+    games = Player.order(updated_at: :asc)
+    render json: games, status: :ok
+  end
+
+  #history of 1 game
+  def leaderboard
+    type = params[:name]
+    if type == nil
+      return render json: {error: "game name required"}, status: 404
```

Comment:
This condition can be checked on one line for cleaner code and I think a 400 Bad Request makes more sense here than a 404 Not Found.
```suggestion
     render json: {error: "game name required"}, status: :bad_request and return if type.nil?
```